src/components/SurveyComponents/GeneralQuestions.tsx

```diff
@@ -0,0 +1,336 @@
+  },
+  workLocation: {
+    type: 'MCQ',
+    question: 'You work from',
+    choices: [
+      'Office for a company in Egypt',
+      'Home for a company in Egypt',
+      'Hybrid for a company in Egypt (e.g 2-3 days from home or office)',
+      'Home for a company abroad (Remote employee)',
+      'Abroad in another country (Relocated elsewhere)',
+    ],
+    isRequired: true,
+  },
+  companySize: {
+    type: 'MCQ',
+    question: 'Your company is considered:',
+    choices: [
+      'A Start-up',
+      'A Small and Medium Enterprise',
+      'A Large Enterprise/Multinational Company',
+    ],
+    isRequired: true,
+  },
+  companyLocationEgypt: {
+    type: 'MCQ',
+    question: "Company Location in case it's operating from Egypt:",
+    choices: [
+      'No site in Egypt',
+      'Cairo',
+      'Alexandria',
+      'Hurghada (Or Gouna)',
+      'Port Said / Suez',
+      'Mansoura',
+      'Faiyum',
+      'Luxor / Aswan',
+      'Other',
+    ],
+    isRequired: true,
+  },
+  companyType: {
+    type: 'MCQ',
+    question: 'You work in a:',
+    choices: ['Software House', 'Product-based company', 'Other'],
+    isRequired: true,
+  },
+  businessFocus: {
+    type: 'MCQ',
+    question: 'The business main focus is:',
+    choices: Object.values(businessLineMap),
+    isRequired: true,
+  },
+  businessScope: {
+    type: 'MCQ',
+    question: 'The business is',
+    choices: [
+      'Local / National (operates inside Egypt)',
+      'Regional (has some business lines in Saudi Arabia, Emirates, etc)',
+      'Global (has some business lines in the region, the US, Europe, Asia, etc.)',
+    ],
+    isRequired: false,
+  },
+  businessIndustries: {
+    type: 'checkbox',
+    question: 'The industries of the business are',
+    choices: [
+      'Fintech',
+      'Healthcare',
+      'Insurance',
+      'Automotive',
+      'Delivery',
+      'Supply Chain',
+      'Transportation',
+      'Telecommunications',
+      'Energy',
+      'Manufacturing',
+      'Technology',
+      'Education',
+      'Human Resources Services',
+      'Law',
+      'Agriculture',
+      'Production',
+      'Aerospace',
+      'Other',
+    ],
+    isRequired: false,
+  },
+}
+
+const engineeringRoles = new Set(
+  POSITIONS_OPTIONS.filter((position) =>
+    ['engineer', 'developer', 'data'].includes(position.category),
+  ).map((pos) => pos.positionName),
+)
+const managementRoles = new Set(
+  POSITIONS_OPTIONS.filter((position) => position.category == 'manager').map(
+    (pos) => pos.positionName,
+  ),
+)
+const productRoles = new Set(
+  POSITIONS_OPTIONS.filter((position) => ['product'].includes(position.category)).map(
+    (pos) => pos.positionName,
+  ),
+)
+// MISC: other, designer
+
+// TODO: clear fields after current page because if user goes general => product => general => engineering the answers for product would still be filled.
+function getRoleSpecificPage(role: string): number {
+  if (role == '') {
+    return 1
```

Comment:
for a better design and for clearer code readability, how about indexing the pages with enum values instead of numbers?? 1, 2, 3, 4 are not descriptive of which page we should redirect to